tests/schemer/schema_test.py

```diff
@@ -210,9 +210,11 @@ def test_set_non_nullable_field_to_none(self):
         self.document_1['external_code'] = None
         self.assert_document_paths_invalid(self.document_1, ['external_code'])
 
-    def test_incorrect_type(self):
+    def test_dynamic_function_exception(self):
         self.document_1['author'] = 33
-        self.assert_document_paths_invalid(self.document_1, ['author'])
+        with self.assertRaises(SchemaFormatException) as cm:
+            blog_post_schema.validate(self.document_1)
+        self.assertEqual('author', cm.exception.path)
```

Comment:
Can we also test what happens when the dynamic function succeeds in returning a type, but the actual value being validated then fails the type check.